app/api/templates/specs/templates.spec.js

```diff
@@ -348,6 +347,37 @@ describe('templates', () => {
           .catch(done.fail);
       });
     });
+
+    describe('generatedId', () => {
+      const populateGeneratedIdByTemplateSpy = jest
+        .spyOn(generatedIdPropertyAutoFiller, 'populateGeneratedIdByTemplate')
+        .mockImplementation(() => Promise.resolve());
+
+      afterEach(() => {
+        populateGeneratedIdByTemplateSpy.mockReset();
+      });
+
+      describe('when there is a new property with generatedId type', () => {
+        it('should call populateGeneratedIdBTemplate to auto-fill values', async () => {
+          const templateToUpdate = {
+            _id: templateToBeEditedId,
+            name: 'template to be edited',
+            commonProperties: [{ name: 'title', label: 'Title', type: 'text' }],
+            properties: [{ name: 'autoId', type: propertyTypes.generatedid, label: 'Auto Id' }],
+          };
+          await templates.save(templateToUpdate, 'en');
+
+          expect(populateGeneratedIdByTemplateSpy).toHaveBeenCalled();
```

Comment:
Can we change to a toHaveBeenCalledWith() to check that contents are correct?

Reply:
Sure, it's done